app/services/katello/pulp/repository.rb

```diff
@@ -120,6 +120,56 @@ def mirror_importer_connection_options
       def proxy_host_importer_value
         root.ignore_global_proxy ? "" : nil
       end
+
+      def refresh
+        repository_details = backend_data
+        update_or_associate_importer(repository_details)
+        update_or_associate_distributors(repository_details)
+        remove_unnecessary_distributors(repository_details)
+      end
+
+      def update_or_associate_importer(repository_details)
+        existing_importers = repository_details["importers"]
```

Comment:
instead of passing in repository_details, you could just call backend_data here every time, its cached after the first request